specs/File Classification Spec.md

```diff
@@ -0,0 +1,59 @@
+# **PowerToys File Classification Spec**
+
+![MS PowerToys](https://hothardware.com/ContentImages/NewsItem/48038/content/Microsoft_PowerToys.jpg "PowerToys")
+## Users can quickly rename and group files.
+### Authors: Benjamin Leverette and Prudence Phillips
+### Spec Status: Draft
+## 1. Overview
+
+**1.1. Elevator Pitch / Narrative** 
+
+David has recently uploaded thousands of pictures from his camera to his Surface Laptop.  Unfortunately, the images all have generic names such as 'IMG_141' instead of labels that he can identify.  David downloads the File Classification PowerToy that allows him to create new folders with files that share a label, rename a group of files with a label, or change the name or portion of a name in a group of files to another name.
+
+**1.2. Customers**
+
+Like all utilities from PowerToys, the File Classification feature is for power users and developers who are looking to tune and streamline their Windows experience for greater productivity.
+  
+**1.3. Problem Statement and Supporting Customer Insights**
+
+Power users need a better way to organize files, from renaming files to creating new folders for similar files.  Our PowerToys Consumer Survey received feedback validating the usefulness of a feature that provides such functionality.
+
+**1.4. Existing Solutions or Expectations**
+
+Users currently have to highlight a group of files and right click to rename them.  Users must manually create a folder, move files to that folder, and rename them.  There are third-party resources that allow users to rename files similarly.
+
+We expect users to install and enable PowerToys for Windows in order to access the File Classification utility.
+
+**1.5. Goals/Non-Goals**
+
+Design and develop a feature that can rename and group files within an 8 week period.
+
+## 2. Definition of Success
+
+**2.1. Expected Impact: Customer, and Technology Outcomes, Experiments + Measures**
+
+Our PowerToys Consumer Survey received an abundant amount of participation and feedback from a community of passionate power users.  This feature will give them the ability to rename, group and organize files in a way that makes their virtual library of files more organized and efficient.  As interns, we would have 8 weeks to complete the project.
+
+## 3. Requirements
+
+**3.1.	Functional Requirements**
+
+- Users must select a group of files by checking each file or using Shift + directional keys.  They must then use a shortcut to open the File Classification utility, which provides three options:
+  - Automates folder creation and movement of user-selected files into that folder after receiving a label determined by the user.
+  - Renames all user-selected files after receiving a label determined by the user.
+  - Identifies a string of characters in the names of user-selected files and changes it to a string determined by the user.
```

Comment:
I'm not sure I understand the exact mechanics here. Can you provide some sort of examples of what happens here?

I mentally understand the concept of labeling and creating folders. But I don't understand how the "label" I'm providing generates the folder name, what is exactly being looked for in the "identification of string", how as a user I would set that pattern, etc.

Reply:
Agreed!

The initial detail is great, but I think we need a clearer picture of what the exact user experience is. Take a look at my comment about user experience writing in the terminate app spec as I think it applies here as well.

Reply:
I think providing examples is a great idea, and I'll try to do it in a way that doesn't sound too repetitive in terms of functionality.  I thought the final bullet covered the use of labels, but I definitely left out the naming of folders.

Also, I'll note that the string of characters that is identified is inputted by the user.  It is essentially a replace function that takes in characters from the user and every time we see that string in a file name, it will be replaced by another string that is also inputted by the user. I believe the image depicts this as well.